docs: Adding Microtica as a deployment option (#3802)
medusajs/medusa: docs/content/deployments/server/deploying-on-microtica.md

Reviewer: Can we remove all images from this file? We are gradually removing images from our docs as they can grow outdated quickly.

- A [GitHub](https://github.com/) account to create a repository to host your backend's codebase.

### Required Tools

- Git’s CLI tool. You can follow [this documentation to learn how to install it for your operating system](../../development/backend/prepare-environment.mdx#git).

---

## Deploy to Microtica

### Step 1: Create/Import a Git repository

Before you can deploy your Medusa backend you need to connect your preferred Git account. Microtica will create a repository on your Git account with a default repo name `medusa-server`. 

If you already have a Medusa backend repository that you want to deploy then you need to perform a couple of changes in your source code. Follow [this tutorial to deploy and existing Medusa Backend app](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#xUBRz).

### Step 2: Configure the Template

The second step provides customization posibilities by configuring environment variables.

Add an application name and the admin credentials that will be used to create an initial admin user with which you can later sign in to your Medusa Admin. 

> Then, choose whether you want a production Medusa Backend environment or a development one. The production template will provision your managed RDS PostgreSQL and Redis instances. On the other hand, the development template will use a local SQLite and a fake Redis instance.

Reviewer: Can you expand the RDS acronym?

Author: done

> Then, choose whether you want a production Medusa Backend environment or a development one. The production template will provision your managed Amazon RDS (Relational Database Service) PostgreSQL and Redis instances. On the other hand, the development template will use a local SQLite and a fake Redis instance.

Reviewer: ```suggestion
Then, choose whether you want a production Medusa Backend environment or a development one. The production template will provision your managed Relational Database Service (RDS) PostgreSQL and Redis instances. On the other hand, the development template will use a local SQLite and a fake Redis instance.
```

---
description: 'Learn step-by-step.'
addHowToData: true
---

# Deploy Your Medusa Backend on Microtica

In this document, you'll learn how to deploy your Medusa backend on your AWS account with Microtica. 

> [Microtica](https://microtica.com) is a cloud-native delivery platform that enables you to deploy infrastructure and applications to your AWS account, while providing actionable insights to help you optimize resources, performance, & costs. 

Reviewer: ```suggestion
## Overview

[Microtica](https://microtica.com) is a cloud-native delivery platform that enables you to deploy infrastructure and applications to your AWS account, while providing actionable insights to help you optimize resources, performance, & costs. 
```

Furthermore, your Medusa backend should be configured to work with PostgreSQL and Redis. You can follow the [Configure your Backend documentation](../../development/backend/configurations.md) to learn how to do that.

### Needed Accounts

- A [Microtica](https://app.microtica.com/) account. Microtica provides a free plan that you can use.
- An [AWS](https://aws.amazon.com/) account that you’ll connect to Microtica to provision the infrastructure.
- A [GitHub](https://github.com/) account to create a repository to host your backend's codebase.

### Required Tools

- Git’s CLI tool. You can follow [this documentation to learn how to install it for your operating system](../../development/backend/prepare-environment.mdx#git).

---

## What will be provisioned on AWS

Since Microtica deploys on your cloud account, here are the resources that the platform is going to provision in the environment.

- VPC, subnets and networking
- Container infrastructure based on Fargate 
- application load balancer
- persistent storage
- S3 bucket
- Postgres database 

> - and Redis (in production mode)

Reviewer: Can we move this section under the new Overview section in the beginning and change the header level to h3 (i.e. `### What will ...`)

It will take around ten minutes for the solution to be deployed on your cloud account.

You can follow the build pipeline in real-time by clicking the "View Logs" button. 

![Building Resources](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f9a465c27cf7090e5f_gxwdXAAmUxHsSyJlgScHuu4m7GiFTNnwW-OZoAhDC8v-Ez3KmsK-OCKfF2Qe2JUtxMyih7jCA7n-hAZnx0fdFjv-TWyDuoAUvftSRE9hqmPB1gVhG6lH0IhXqbliou1VnAVDvlCtHobzlUrrDhnCmzob9PfdNGrzlhwsL60QsAXJm1gIPE27J6P_Y2Y3.png)

Once the build process is complete, a new deployment with the infrastructure resources is triggered. You can follow the logs of the deployment process by clicking the "View deployment" button, and then selecting the deployment from the list. 

![Deployment Logs](https://global-uploads.webflow.com/62b04b3a715c202b5fa1408b/639727f93aa3974438e56689_vF-4U1L5B_sQQJdrjaE_bdlN2hdTsOcn5eWVRfgzF1UqNAuWAqN2zh9ttWp6SW2jjCqPUNLpRsL4j01FR5emAMcJqCbfwbaBv3Np49JH6hpggYm3rmY-xYCOg8H0dooKnC4x_sTD-JreveLzUxBMxQI3M0gCm-8B2YJ5eJj7dU8mf8qlSTYmnpnExIvB.png)

---

> ## Preview the environment

Reviewer: We recently embedded the admin plugin as part of the medusa backend. I'm not sure if you've tested it, but if it works can you add a section similar to [here](https://docs.medusajs.com/deployments/server/deploying-on-railway#testing-the-admin) on how to test it?

if it doesn't work with the embedded admin then you need to add a note about it as well.

Also change the header to the following:
```suggestion
## Test the Backend
```

After the deployment is finished, navigate to Resources → [AppName] (Medusa in this example) → Overview. Then, under the Resource Outputs section you should see the AccessUrl. This is the backend's URL that you can use to access API endpoints. You can try getting the list of products using the endpoint `/store/products`.

---

> ## Add environment variables

Reviewer: ```suggestion
## Add Environment Variables
```

The environment variables can be updated, added, or configured after deployment as well. You can access them with `process.env.<VARIABLE NAME>`.

You can read more about the built-in environment variables, as well as how to specify custom environment variables in the Medusa backend runtime [in Microtica documentation](https://docs.microtica.com/medusa-server?utm_source=medusa&utm_medium=docs&utm_campaign=medusa#z8li6). 

---

> ## Automated deployments

Reviewer: ```suggestion
## Updating your Deployed Backend
```

> Any updates to the backend (e.g. updating the Medusa core version) are deployed automatically when changes are committed to the repository. Microtica will handle the entire process of building and deploying your application on your connected AWS account.

Reviewer: ```suggestion
Any updates to the backend (for example, updating the Medusa core version) are deployed automatically when changes are committed to the repository. Microtica will handle the entire process of building and deploying your application on your connected AWS account.
```

> 

Reviewer: Remove these additional lines (keep one empty line at the end of the file)
